From 78077a5e80c9ad5f909037a48100481ddfedc6b2 Mon Sep 17 00:00:00 2001
From: Andrey Utkin <andrey_utkin@gentoo.org>
Date: Wed, 13 Dec 2017 01:50:03 +0000
Subject: [PATCH] tests: pass all env vars to cmd-runner

cmd-runner was started with no environment variables inherited.
This breaks tests when run with custom PYTHONPATH, which is useful for
testing botocore while not being installed in standard locations.

One case when this is important is performing tests before installing
the package in Gentoo Linux.

Link: https://bugs.gentoo.org/640726
---
 tests/__init__.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/__init__.py b/tests/__init__.py
index 74a2e4de..fa0b819b 100644
--- a/tests/__init__.py
+++ b/tests/__init__.py
@@ -145,6 +145,7 @@ class BaseClientDriverTest(unittest.TestCase):
         if self.INJECT_DUMMY_CREDS:
             env = {'AWS_ACCESS_KEY_ID': 'foo',
                    'AWS_SECRET_ACCESS_KEY': 'bar'}
+        env.update(os.environ)
         self.driver.start(env=env)
 
     def cmd(self, *args):
-- 
2.15.1

